Projects/Claims/Sources/Views/submitClaim/Checkout/SubmitClaimCheckoutTransferringScreen.swift

```diff
@@ -3,169 +3,238 @@ import hCore
 import hCoreUI
 
 public struct SubmitClaimCheckoutTransferringScreen: View {
-    @State var hasActionCompleted: Bool = false
+    @PresentableStore var store: ClaimsStore
+    @State var loadingAnimation: Bool = false
+    @State var successAnimation: Bool = false
+    @State var errorAnimation: Bool = false
 
+    @Namespace private var animation
     public init() {}
 
     public var body: some View {
-        LoadingViewWithContent(.claimNextSummary) {
-            BlurredProgressOverlay {
-                ZStack(alignment: .center) {
-                    VStack {
-                        Spacer()
-                            .scaleEffect(
-                                x: hasActionCompleted ? 2 : 1,
-                                y: hasActionCompleted ? 2 : 1,
-                                anchor: .center
-                            )
-                        Spacer()
-                    }
-                    .opacity(hasActionCompleted ? 0 : 1)
-                    .animation(.spring(), value: hasActionCompleted)
-
-                    VStack {
-                        Spacer()
-
-                        VStack(spacing: 16) {
-
-                            hText(L10n.Claims.Payout.Progress.title, style: .title2)
-
-                        }
-                        .frame(maxHeight: .infinity, alignment: .top)
+        BlurredProgressOverlay {
+            ZStack(alignment: .center) {
+                VStack {
+                    Spacer()
                         .scaleEffect(
-                            x: hasActionCompleted ? 1 : 0.3,
-                            y: hasActionCompleted ? 1 : 0.3,
-                            anchor: .top
+                            x: loadingAnimation ? 2 : 1,
+                            y: loadingAnimation ? 2 : 1,
+                            anchor: .center
                         )
+                    Spacer()
+                }
+                .opacity(loadingAnimation ? 0 : 1)
+                .animation(.spring(), value: loadingAnimation)
+                LoadingViewWithState(.claimNextSingleItemCheckout) {
+                    successView()
+                } onLoading: {
+                    loadingView()
+                } onError: { error in
+                    errorView(withError: error)
+                }
+            }
+        }
+    }
 
-                        Spacer()
+    @ViewBuilder
+    private func successView() -> some View {
```

Comment:
I usually feel like this is a bit of an anti pattern, makes more sense to do a new struct that implements View imo. But feel free to ignore if you don't feel so 😄